pkg/networkserver/mac/adr.go

```diff
@@ -583,6 +583,13 @@ func adrAdaptDataRate(
 ) float32 {
 	currentParameters, desiredParameters := macState.CurrentParameters, macState.DesiredParameters
 	currentDataRateIndex := currentParameters.AdrDataRateIndex
+	// Keep the data rate index between min and max
+	if currentDataRateIndex < minDataRateIndex {
+		currentDataRateIndex = minDataRateIndex
+	}
+	if currentDataRateIndex > maxDataRateIndex {
+		currentDataRateIndex = maxDataRateIndex
+	}
```

Comment:
We can also do this by using the `min` and `max` built-in functions.
```suggestion
	currentDataRateIndex = max(currentDataRateIndex, minDataRateIndex)
	currentDataRateIndex = min(currentDataRateIndex, maxDataRateIndex)
```

Reply:
Thanks, applied your suggestion.